Zoom and drag for `pl-image-capture` image preview

miguelaenlle opened the pull request.

reteps reviewed (changes requested):
As discussed on call with @miguelaenlle , I am worried about pulling in a small library just for pan/zoom. He said he will give cropperjs a shot for doing the same thing, and we can reevaluate if the implementation gets too complex.

miguelaenlle commented:
As discussed on Slack, I added a few more adjustments:
- The zoom in and out buttons are disabled at the maximum and minimum zoom scale, respectively. 

https://github.com/user-attachments/assets/60aea6b8-1ed8-4a93-809d-21218a67f0d5

- Added grab/grabbing cursor icons in place of the move icon.

miguelaenlle commented on apps/prairielearn/elements/pl-image-capture/pl-image-capture.js:
```diff
@@ -515,6 +521,75 @@
+          });
+
+          let zoomEnabled = false;
+          capturePreview.addEventListener('panzoomzoom', (e) => {
+            const scale = e.detail.scale;
+            zoomEnabled = scale > 1;
+            capturePreview.style.cursor = zoomEnabled ? 'grab' : 'default';
+
+            // Only when zoomed in, indicate that panning is available.
+            // Panzoom has an option called panOnlyWhenZoomed, but it does not update the cursor.
+            if (scale === MIN_ZOOM_SCALE) {
+              zoomOutButton.classList.add('disabled', 'opacity-10');
+            } else {
+              zoomOutButton.classList.remove('disabled', 'opacity-10');
+            }
+
+            if (scale >= MAX_ZOOM_SCALE) {
+              zoomInButton.classList.add('disabled', 'opacity-10');
+            } else {
+              zoomInButton.classList.remove('disabled', 'opacity-10');
+            }
+          });
+
+          capturePreview.addEventListener('panzoomstart', () => {
+            if (zoomEnabled) {
+              capturePreview.style.cursor = 'grabbing';
+            } else {
+              capturePreview.style.cursor = 'default';
+            }
+          });
+
+          capturePreview.addEventListener('panzoomend', () => {
+            if (zoomEnabled) {
+              capturePreview.style.cursor = 'grab';
+            }
+          });
+        } else {
+          this.imageCapturePreviewPanzoom.reset({ animate: false });
+        }
+      }
```
We want to throw if Panzoom initialization fails: that is an issue.

miguelaenlle commented on apps/prairielearn/elements/pl-image-capture/pl-image-capture.js:
```diff
@@ -515,6 +521,75 @@
+            this.imageCapturePreviewPanzoom.zoomOut();
+          });
+
+          let zoomEnabled = false;
+          capturePreview.addEventListener('panzoomzoom', (e) => {
+            const scale = e.detail.scale;
+            zoomEnabled = scale > 1;
+            capturePreview.style.cursor = zoomEnabled ? 'grab' : 'default';
+
+            // Only when zoomed in, indicate that panning is available.
+            // Panzoom has an option called panOnlyWhenZoomed, but it does not update the cursor.
+            if (scale === MIN_ZOOM_SCALE) {
+              zoomOutButton.classList.add('disabled', 'opacity-10');
+            } else {
+              zoomOutButton.classList.remove('disabled', 'opacity-10');
+            }
+
+            if (scale >= MAX_ZOOM_SCALE) {
+              zoomInButton.classList.add('disabled', 'opacity-10');
+            } else {
+              zoomInButton.classList.remove('disabled', 'opacity-10');
+            }
+          });
+
+          capturePreview.addEventListener('panzoomstart', () => {
+            if (zoomEnabled) {
+              capturePreview.style.cursor = 'grabbing';
+            } else {
+              capturePreview.style.cursor = 'default';
+            }
+          });
+
+          capturePreview.addEventListener('panzoomend', () => {
+            if (zoomEnabled) {
+              capturePreview.style.cursor = 'grab';
+            }
+          });
```
I don't think this is an issue because they're added only when `imageCapturePreviewPanzoom` is null, and `imageCapturePreviewPanzoom` will not be null again after it is set. Thus, the event listeners will only be added once. Also, `imageCapturePreviewPanzoom` will remain in use until the user leaves the page.
